cmd/kubeadm/app/apis/kubeadm/bootstraptokenhelpers.go

```diff
@@ -61,13 +61,13 @@ func encodeTokenSecretData(token *BootstrapToken, now time.Time) map[string][]by
 	if token.Expires != nil {
 		// Format the expiration date accordingly
 		// TODO: This maybe should be a helper function in bootstraputil?
-		expirationString := token.Expires.Time.Format(time.RFC3339)
+		expirationString := token.Expires.Time.UTC().Format(time.RFC3339)
 		data[bootstrapapi.BootstrapTokenExpirationKey] = []byte(expirationString)
 
 	} else if token.TTL != nil && token.TTL.Duration > 0 {
 		// Only if .Expires is unset, TTL might have an effect
 		// Get the current time, add the specified duration, and format it accordingly
-		expirationString := now.Add(token.TTL.Duration).Format(time.RFC3339)
+		expirationString := now.Add(token.TTL.Duration).UTC().Format(time.RFC3339)
```

Comment:
i think, the bootstraptoken string format must match between the kubeadm generated tokens (as secrets) and the bootstrap signer controller. would the controller be able to parse this secret time format?

Reply:
https://github.com/kubernetes/kubernetes/blob/c4df930252fab1a13deaa72ef97173616a4e193f/cmd/kubeadm/app/apis/kubeadm/bootstraptokenhelpers_test.go#L30
It should be feasible. Because in the unit test, our time zone is set to utc

Reply:
this string is written in a token / secret:
https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/#bootstrap-token-secret-format

if the bootstrapsigner controller assumes non-UTC, but the token uses UTC, this is a problem.
i still think we should remove this change.

Reply:
> this string is written in a token / secret:
> https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/#bootstrap-token-secret-format
> 
> if the bootstrapsigner controller assumes non-UTC, but the token uses UTC, this is a problem.
> i still think we should remove this change and only modify output visible to users.

I understand your concern, as it has been deposited into the etcd, so it may pose a risk